### PR TITLE
Add test for keylime agent without mTLS

### PR DESCRIPTION
**Pull request by ansasaki**
Add function/basic-attestation-without-mtls test case.
The test case runs a basic attestation using the agent with mTLS disabled.

### CODE REVIEW

**Comment by kkaarreell** [comment]
Hi @ansasaki , do you know if mTLS comes into play when revocation scripts from payload comes into play? If not, we could simplify the test a bit by not using payload (and do not need expect).

File: functional/basic-attestation-without-mtls/test.sh [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            # start tpm emulator
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            # make sure tpm2-abrmd is running
+            rlServiceStart tpm2-abrmd
+            sleep 5
+            # start ima emulator
+            export TPM2TOOLS_TCTI=tabrmd:bus_name=com.intel.tss2.Tabrmd
+            # workaround for https://github.com/keylime/rust-keylime/pull/286
+            export PATH=/usr/bin:$PATH
```
Comment (kkaarreell) [inline functional/basic-attestation-without-mtls/test.sh:29]: Please remove this workaround, I believe it is not needed anymore.

**Comment by ansasaki** [comment]
> Hi @ansasaki , do you know if mTLS comes into play when revocation scripts from payload comes into play? If not, we could simplify the test a bit by not using payload (and do not need expect).

For now, the execution of revocation scripts are not affected by mTLS disablement, but the initial payload script execution is restricted. Note that usually the revocation actions are configured/controlled by the initial payload script, so it is indirectly affected.

When mTLS is disabled, the initial payload script is executed only if `enable_insecure_payload` is set as `True`. I want to exercise this restriction, but probably I need to change the test to not set `enable_insecure_payload` and expect the payload to not be executed, then enable it and expect the payload to be executed.

The python agent is missing the [enhancement 55](https://github.com/keylime/enhancements/blob/master/55_revocation_actions_without_python.md), which adds the `allow_payload_revocation_actions` option to control whether the payload revocation actions should be executed or only pre-installed actions are enabled. This behavior should be tested in a separate test executed only for the rust agent (probably the same used to test if non-python scripts).

File: functional/basic-attestation-without-mtls/test.sh (hunk 1 above)
Comment (ansasaki) [reply functional/basic-attestation-without-mtls/test.sh:29]: Will do.

**Comment by kkaarreell** [comment]
This is weird. CentOS Stream 8 test should be passing as it is running with upstream bits but it fails in a same way as Fedora runs with distribution rpms. @ansasaki Any idea why?

**Comment by ansasaki** [comment]
> This is weird. CentOS Stream 8 test should be passing as it is running with upstream bits but it fails in a same way as Fedora runs with distribution rpms. @ansasaki Any idea why?

This is weird indeed, for me the tests should be passing on both upstream and Fedora since the exercised changes are already there. I'll investigate.

**Comment by kkaarreell** [comment]
OK. Thank you.
Also, could you please add a bit more detailed description to `main.fmf` file? Something similar to what is currently being added in https://github.com/RedHat-SP-Security/keylime-tests/pull/92/files

File: functional/basic-attestation-without-mtls/test.sh [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        rlRun "expect add.expect"
+        rlRun "limeWaitForAgentStatus $AGENT_ID 'Get Quote'"
+        rlRun -s "lime_keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'$AGENT_ID'" $rlRun_LOG -E
+        rlAssertGrep "mTLS disabled" $(limeAgentLogfile)
+        rlAssertGrep "payloads cannot be deployed" $(limeAgentLogfile)
+    rlPhaseEnd
+
+    rlPhaseStartTest "Enable insecure payload and set script_payload"
+        rlRun -s "lime_keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u $AGENT_ID -c delete"
```
Comment (kkaarreell) [inline functional/basic-attestation-without-mtls/test.sh:89]: Adding `lime_keylime_tenant -r 127.0.0.1 -t 127.0.0.1 -u d432fbb3-d2f1-4a97-9ef7-75bd81c00000 -c regdelete` below fixes the CI failure.

Comment (ansasaki) [reply functional/basic-attestation-without-mtls/test.sh]: Rebased on top of current `main` and added this.

**Comment by ansasaki** [comment]
> OK. Thank you. Also, could you please add a bit more detailed description to `main.fmf` file? Something similar to what is currently being added in https://github.com/RedHat-SP-Security/keylime-tests/pull/92/files

Done!

**Comment by kkaarreell** [comment]
I think I will disable coverage measurement on CentOS Stream 8, this particular error already appeared in the past on this platform.
Thank you for the test. Please remember that in order to enable it in upstream keylime/keylime project one has to add the test to the packit test plan.

**Pull request merged**